send_email_messages/frontend/index.js

```diff
@@ -0,0 +1,189 @@
+    const volunteers = delivery.getCellValue("Assigned volunteer");
+    if (volunteers && volunteers.length > 0) {
+      volunteer = findById(allVolunteers, volunteers[0].id);
+    }
+
+    for (let messageType of Object.keys(EMAIL_TYPES)) {
+      if (existingMessageTypes.includes(messageType)) {
+        // We've already sent a message of this type for this delivery
+        continue;
+      }
+      if (EMAIL_TYPES[messageType].stage != deliveryStageId){
+        // The delivery is in the wrong stage for this type of email, skip
+        continue;
+      }
+
+      const {message, warning} = await formatMessageRecord(messageType, delivery, volunteer, EMAIL_TYPES[messageType]["audience"]);
+      if (message) {
+        output.messagesToCreate.push(message);
+      }
+      if (warning) {
+        output.warnings.push(warning);
+      }
+    }
+  }
+  return output;
+}
+
+
+function WarningAccordion({warnings}) {
+
+  if (warnings.length === 0) {
+    return null;
+  }
+  return (
+    <Accordion title={ `${warnings.length} warnings`}>
+      {warnings.map((warning, i) => {
+        return <p key={`warning_${i}`}>{warning}</p>
+      })}
+    </Accordion>
+  )
+}
+
+function MainUIComponent() {
+  const [step1result, setStep1result] = useState(undefined);
+  const [warnings, setWarnings] = useState([]);
+
+  function addWarning(newWarning) {
+    if (newWarning) {
+      setWarnings(warnings.concat(newWarning));
+    }
+  }
+
+  async function refreshQueue() {
+    var result = ''
+
+    // Clear out the warnings from any previous run
+    setWarnings([]);
+    let {messagesToCreate, warnings} = await computeMessagesToCreate();
+    setWarnings(warnings);
+
+    if (messagesToCreate.length == 0) {
+      setStep1result(`No new messages to enqueue.`);
+      return;
+    }
+    if (messagesToCreate.length > 50) {
+      addWarning(`You can only create 50 emails at once, but we found ${messagesToCreate.length} Doing the first 50 now... remember to do this again to get the rest!`);
+      messagesToCreate = messagesToCreate.slice(0, 50);
+    } 
+    await base.getTable("Messages").createRecordsAsync(messagesToCreate);
+    setStep1result(`**Success!** Enqueued ${messagesToCreate.length} new messages`);
+  }
+
+  return (
+    <Box padding={3}>
+      <h2> Step 1: Refresh Queue </h2>
+      <p> Open the Messages tab to get started. Click below to refresh queue (won't send emails yet). </p>
+      <Button variant="primary" onClick={refreshQueue}>
+        Refresh queue
+      </Button>
+      <WarningAccordion warnings={warnings}/>
+      {step1result && <>
+        <p> {step1result} </p>
+        <SendMessagesStep/>
+      </>}
+    </Box>
+  );
+}
+
+function ComponentWithSettings() {
+  const [isShowingSettings, setIsShowingSettings] = useState(false);
+  useSettingsButton(function() {
+    setIsShowingSettings(!isShowingSettings);
+  });
+  
+  if (isShowingSettings) {
+    return <SettingsComponent exit={() => setIsShowingSettings(false)}/>
```

Comment:
Dumb question: how do I access the settings component? Couldn't figure out how to do it from the UI